src/analysis/von_neumann_algebra/basic.lean

```diff
@@ -0,0 +1,84 @@
+/-
+Copyright (c) 2022 Scott Morrison. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Scott Morrison
+-/
+import analysis.normed_space.dual
+import analysis.normed_space.star.basic
+import analysis.complex.basic
+import analysis.inner_product_space.adjoint
+
+/-!
+# Von Neumann algebras
+
+We give the "abstract" and "concrete" definitions of a von Neumann algebra.
+We still have a major project ahead of us to show the equivalence between these definitions!
+
+An abstract von Neumann algebra `wstar_algebra M` is a C^* algebra with a Banach space predual,
+per Sakai (1971).
+
+A concrete von Neumann algebra `von_neumann_algebra H` (where `H` is a Hilbert space)
+is a *-closed subalgebra of bounded operators on `H` which is equal to its double commutant.
+
+We'll also need to prove the von Neumann double commutant theorem,
+that the concrete definition is equivalent to a *-closed subalgebra which is weakly closed.
+-/
+
+universes u v
+
+/--
+Sakai's definition of a von Neumann algebra as a C^* algebra with a Banach space predual.
+
+So that we can unambiguously talk about these "abstract" von Neumann algebras
+in parallel with the "concrete" ones (weakly closed *-subalgebras of B(H)),
+we name this definition `wstar_algebra`.
+-/
+class wstar_algebra (M : Type u) [normed_ring M] [star_ring M] [cstar_ring M]
+  [module ℂ M] [normed_algebra ℂ M] [star_module ℂ M] :=
+(exists_predual : ∃ (X : Type u) [normed_group X] [normed_space ℂ X] [complete_space X],
+  nonempty (normed_space.dual ℂ X ≃ₗᵢ⋆[ℂ] M))
+
+open_locale inner_product
+
+/-- The commutant of a subsemiring. -/
+def subsemiring.commutant {A : Type u} [semiring A] (B : subsemiring A) : subsemiring A :=
+{ carrier := { x : A | ∀ y ∈ B, x * y = y * x },
```

Comment:
This is what we call `centralizer` in #11946 I think

Reply:
Ah, good point. My preference would be that we call it `centralizer` at the monoid/group level, and then `commutant` at the ring/algebra level. It's a slight inconsistency, but well attested in the literature!

Reply:
If someone wants to hit merge on #11946 (LGTM!) I'll adapt this on top of that before continuing further.

Reply:
Note that we previously used `centralizer` for rings too before we accidentally deleted everything we had about centralizer:

https://github.com/leanprover-community/mathlib/blob/aa95e33d46a70323431339213461e759e9833f97/src/algebra/commute.lean#L275

Reply:
No, you're right. In purely algebraic settings people use "centralizer" for rings too. It's only in the functional analysis world that we switch to "commutant".

Reply:
We should probably have a discussion about this on Zulip, because this is not going to be the only oddity in naming in the realm of operator algebras. For example, a simple von Neumann algebra is called a *factor*. It would be good to get community input before we pick an approach.

Reply:
https://leanprover.zulipchat.com/#narrow/stream/116395-maths/topic/naming.20in.20operator.20algebras

Reply:
Naming aside, we should implement this as
```suggestion
{ carrier := set.centralizer (B : set A),
```
so that we can reuse results about `set.centralizer` in the proofs below.